### DISCUSSION TITLE
How to speed up "selected worker" stage?

### DISCUSSION THREAD

**Discussion by patmaddox** [concourse/concourse#6908] [Help & Support] [answered]
I believe this is similar to #6860 but I didn't want to hijack it.

Our jobs are taking a total of 3 minutes right now. The first 2 minutes are spent in:

```
selected worker: concourse-worker-0
initializing
selected worker: concourse-worker-0
selected worker: concourse-worker-0
selected worker: concourse-worker-0
```

and then it only takes a minute to run. There are four "selected worker" messages, and the last one takes up most of the time.

I've currently got it configured for only 1 worker. I would like to find a way to speed up that part if possible. But, I don't even know what it's doing - I've enabled `--verbose` and there's no output for those specific parts.

So how could I find out what it's doing? And how might I speed it up?

**Comment by HemangLavana** [comment]
I am facing the same issue. Would like to know how to speed up this part.

**Comment by patmaddox** [comment, marked as answer]
It appears that setting `worker.baggageclaim.driver="detect"` did the trick.

**Reply by HemangLavana** [reply]
thx, will give it a try

**Reply by taylorsilva** [reply]
More detailed why: the default driver in the chart is `naive` https://github.com/concourse/concourse-chart/blob/c17bd3994507e3a3600fb2c6e870f28d7d1ba6ef/values.yaml#L1725-L1728
naive literally does `cp volume1 volume2` for every step in a pipeline, which ends up taking a long time. The larger your image and inputs are, the longer everything will take. We actually recommend using `overlay` which is probably what `detect` figured out to use.

Seems like this value was initially set to naive when it was first introduced. Seems to me it would make sense for it to have a default value of detect (or none and just do what the Concourse binary wants to default to).

**Reply by taylorsilva** [reply]
https://github.com/concourse/concourse-chart/pull/248
